### PR TITLE
Audio overhaul

### PR DESCRIPTION
**Pull request by tomdixon**
Changes to support single and multiple instances of sound clips playing predictably. 
Also supporting audio pause fade time.

### CODE REVIEW

File: src/hooks/useAudioPlayer.ts [hunk 1]
```diff
@@ -2,221 +2,309 @@ import { Callbacks, CogsClientMessage } from '@clockworkdog/cogs-client';
...
+            volume,
+          };
+
+          // Start fade when clip starts
+          if (isFadeValid(fade)) {
+            clip.player.volume(0, soundId);
+            clip.player.once(
+              'play',
+              () => {
+                // clip.player.once('fade', () => clip.player.volume(volume, soundId), soundId);
```
Comment (chetbox) [inline src/hooks/useAudioPlayer.ts:180]: Why the change from `once` here?

File: src/hooks/useAudioPlayer.ts [hunk 2]
```diff
@@ -2,221 +2,309 @@ import { Callbacks, CogsClientMessage } from '@clockworkdog/cogs-client';
...
-            clip.player.fade(clip.player.volume(), 0, fade * 1000);
+    (path: string, { fade }: { fade?: number }) => {
+      const clipPlayer = audioClipPlayers[path];
+      if (!clipPlayer) {
+        return;
+      }
+      const { player, activeClips } = clipPlayer;
+
+      // Cleanup any old fade callbacks first
+      player.off('fade');
```
Comment (chetbox) [inline src/hooks/useAudioPlayer.ts:207]: 👍🏼

**Comment by chetbox** [comment]
@tomdixon Have you got your audio project pack so I can put it through its paces? I'm sure you've tested loads but probably good to exercise the code some more.

**Comment by tomdixon** [comment]
> @tomdixon Have you got your audio project pack so I can put it through its paces? I'm sure you've tested loads but probably good to exercise the code some more.

It's in this PR I've just opened https://github.com/clockwork-dog/cogs/pull/649

File: src/hooks/useAudioPlayer.ts (hunk 1 above)
Comment (tomdixon) [reply src/hooks/useAudioPlayer.ts:180]: This was redundant, the `fade` does indeed change the internal Howl player's volume. I'll remove the line.

**Review by chetbox — APPROVED** [review]
The acid tests project is really good. Covers so many cases! Really helpful for testing this.
